scalacheck/src/test/scala/karat/scalacheck/TestCounter.scala

```diff
@@ -64,11 +65,35 @@ object TestCounter extends Properties("Sample") {
     }
 
   val initialState: Int = 0
-  val stepAction: (Action, Int) => Step[Int, Int] = right
+  val stepAction: (Action, Int) => Option[Step[Int, Int]] = right
   val initialFormula: Formula[Info[Action, Int, Int]] = formula
 
-  property("checkRight") = forAll(model.gen) { actions =>
-    checkFormula(actions, initialState, stepAction)(initialFormula)
+  // TODO
+  // Maybe this is provided by scalacheck-effect
+  implicit class PropFOps[F[_]](effectProp: F[Prop.Result]) {
```

Comment:
@fedefernandez According to [the source](https://github.com/typelevel/scalacheck-effect/blob/main/core/shared/src/main/scala/org/scalacheck/effect/PropF.scala#L37-L40):

```
  * [...] When computing the
  * result of a single sample, `F[Unit]` values are treated as successes and any exceptions thrown
  * are treated as falsifications.
```

This seems to be a departure of how failures are represented in ScalaCheck.

Reply:
I'm not sure if I'm following you. Here we have a `F[Prop.Result]` and we want to generate a `PropF[F]`.